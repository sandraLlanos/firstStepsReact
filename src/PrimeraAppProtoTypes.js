import React from 'react';
import PropTypes from 'prop-types';

const PrimeraAppProtoTypes = ( {saludo, subtitulo } ) => { 
  //  if (!saludo){
  //     throw new Error('el saludo es necesario'); 
  //  }
   
    return (
       <>
     
          <h1>{ saludo }</h1>
          <p>{ subtitulo }</p>
          
       </>
    );
 }

//  con propTypes podemos hacer la validacion del tipo de 
// dato de la propiedad y tambi√©n si es requerida o no 
 PrimeraAppProtoTypes.propTypes ={
   saludo:PropTypes.string.isRequired
 }
 PrimeraAppProtoTypes.defaultProps = {
   subtitulo: 'Este es el subtitulo por defecto'
 }
 export default PrimeraAppProtoTypes;