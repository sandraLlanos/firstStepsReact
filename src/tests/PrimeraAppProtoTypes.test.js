import React from 'react';
import { shallow } from 'enzyme';
import PrimeraAppProtoTypes from '../PrimeraAppProtoTypes';

describe('prueba sobre <PrimeraAppProtoTypes />', () => {
    
    test('debe de mostrar <PrimeraAppProtoTypes /> correctamete ', () => {
        const saludo = 'Hola soy Sandra';
        const wrapper =  shallow( <PrimeraAppProtoTypes saludo={saludo}/> );

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola soy Sandra';
        const subtitulo = 'Este no es un subtitulo por defecto';
        const wrapper =  shallow( 
            <PrimeraAppProtoTypes 
                saludo={saludo}
                subtitulo={subtitulo}
                />
         );
         //find funciona como el document.querySelector de js
        const textoparrafo = wrapper.find('p').text();
        console.log(textoparrafo)
        expect( textoparrafo ).toBe(subtitulo);
    })
    
    
})
