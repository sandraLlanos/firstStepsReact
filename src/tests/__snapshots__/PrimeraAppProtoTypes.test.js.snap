// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba sobre <PrimeraAppProtoTypes /> debe de mostrar <PrimeraAppProtoTypes /> correctamete  1`] = `
<Fragment>
  <h1>
    Hola soy Sandra
  </h1>
  <p>
    Este es el subtitulo por defecto
  </p>
</Fragment>
`;

exports[`prueba sobre <PrimeraAppProtoTypes /> debe de mostrar el subtitulo enviado por props 1`] = `
<Fragment>
  <h1>
    Hola soy Sandra
  </h1>
  <p>
    Este no es un subtitulo por defecto
  </p>
</Fragment>
`;
